doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+<article class="markdown-body">
+
+High level requirements
+=======================
+
+<nb>Some requirements are only motivated by the fact that the reference
+implementation implements them. Search for "reference implementation".</nb>
+
+# Introduction
+The PF **MUST** be a library.
+<why>To be reused in different components.</why>
+
+PF instances **MUST NOT** share any mutable data.
+<why>to guarantee that different PF instances will not impact each others.</why>
+
+# Parameters
+
+A PF **MUST** be able to handle parameters.
+<why>because the PF aims to abstract hardware and model it by parameters.</why>
```

Comment:
what is a parameter? Is there a link between this req and any description of the aim of the PFW? (what problem is the PFW supposed to solve?)